packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -686,4 +693,120 @@ export class AbstractQueryInterfaceTypeScript {
   ): Promise<void> {
     await this.sequelize.queryRaw(this.queryGenerator.getToggleForeignKeyChecksQuery(enable), options);
   }
+
+  /**
+   * Create a new savepoint
+   *
+   * @param transaction
+   * @param options
+   */
+  async createSavepoint(transaction: Transaction, options: CreateSavepointOptions): Promise<void> {
+    if (!this.queryGenerator.dialect.supports.savepoints) {
+      throw new Error('Savepoints are not supported for the current dialect!');
+    }
+
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to create a savepoint without transaction object!');
+    }
+
+    const sql = this.queryGenerator.createSavepointQuery(options.savepointName);
+    await this.sequelize.queryRaw(sql, { ...options, transaction, supportsSearchPath: false });
```

Comment:
The interface `CreateSavepointOptions` should omit `transaction`, `connection` and `supportsSearchPath` that it inherits from `QueryRawOptions`

Same for `StartTransactionOptions`, `CommitTransactionOptions`, `RollbackSavepointOptions`, `RollbackTransactionOptions`, `SetIsolationLevelOptions`

Reply:
Done